test/server/cards/01_SOR/events/TheEmperorsLegion.spec.ts

```diff
@@ -0,0 +1,55 @@
+describe('The Emprerors Legion', function () {
+    integration(function (contextRef) {
+        it('should return card to player hand from a discard pile', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['the-emperors-legion'],
+                    groundArena: ['pyke-sentinel', 'seasoned-shoretrooper', 'blizzard-assault-atat'],
+                    spaceArena: ['cartel-spacer'],
+                    discard: ['keep-fighting', 'green-squadron-awing', 'disarm']
+                },
+                player2: {
+                    hand: ['superlaser-blast'],
+                    discard: ['tactical-advantage', 'guerilla-attack-pod']
+                }
+            });
+            const { context } = contextRef;
+
+            context.player1.passAction();
+            context.player2.clickCard(context.superlaserBlast);
+            expect(context.pykeSentinel.location).toBe('discard');
+            expect(context.seasonedShoretrooper.location).toBe('discard');
+            expect(context.blizzardAssaultAtat.location).toBe('discard');
+            expect(context.cartelSpacer.location).toBe('discard');
+            context.player1.clickCard(context.theEmperorsLegion);
```

Comment:
Do you mind breaking this into shorter code blocks for readability, 4 - 6 lines to a block is ideal